Sources/TSCUtility/Version.swift

```diff
@@ -109,7 +109,8 @@ public extension Version {
     ///
     /// - Parameters:
     ///   - string: The string to parse.
-    init?(string: String) {
+    ///   - lenient: Boolean indicating if partial strings should be parsed. Infers that strings without an explicit patch version have a patch version of zero. Defaults to false.   
+    init?(string: String, lenient: Bool = false) {
```

Comment:
since there are some dependencies in SwiftPM on the concrete ctor signature, lets structure this as:

```
init?(string: String) {
 self.init(string: string, lenient: false)
}

init?(string: String, lenient: Bool) {
 ...
}
```

Reply:
Do we really need the the `lenient` parameter? If the goal is to treat all x.y as x.y.0, then I don't we can use strict parsing anywhere.

Reply:
I'm happy to remove the lenient option if no one needs it as a solution.

Reply:
imo its good to have this option for now so we can enable/disable it from SwiftPM side based on which tools version we are targeting - as this change could break existing packages. @neonichu wdyt?

Reply:
I agree that the flag is useful.

Reply:
> imo its good to have this option for now so we can enable/disable it from SwiftPM side based on which tools version we are targeting - as this change could break existing packages.

This makes sense. I keep forgetting that SwiftPM needs to support previous versions.

Reply:
Thanks for the feedback and pointing me to exactly what the test failure was. I've made the changes and run the in-module tests locally.

Reply:
>  I keep forgetting that SwiftPM needs to support previous versions.

that's the hard part :D